partitioner/src/test/java/io/confluent/connect/storage/partitioner/TimeBasedPartitionerTest.java

```diff
@@ -0,0 +1,180 @@
+        DateTimeFormatter formatter = DateTimeFormat.forPattern(pathFormat).withZone(DATE_TIME_ZONE);
+        long utc1 = DATE_TIME_ZONE.convertLocalToUTC(time.getMillis() - TimeUnit.MINUTES.toMillis(60), false);
+        long utc2 = DATE_TIME_ZONE.convertLocalToUTC(time.getMillis() - TimeUnit.MINUTES.toMillis(120), false);
+        DateTime time1 = new DateTime(DATE_TIME_ZONE.convertUTCToLocal(utc1));
+        DateTime time2 = new DateTime(DATE_TIME_ZONE.convertUTCToLocal(utc2));
+        assertEquals(time1.toString(formatter), time2.toString(formatter));
+    }
+
+    @Test
+    public void testRecordFieldTimeExtractor() throws Exception {
+        TimeBasedPartitioner<String> partitioner = new TimeBasedPartitioner<>();
+        Map<String, Object> config = createConfig("timestamp");
+        partitioner.configure(config);
+
+        long timestamp = new DateTime(2015, 4, 2, 1, 0, 0, 0, DateTimeZone.forID(timeZoneString)).getMillis();
+        SinkRecord sinkRecord = createSinkRecord(timestamp);
+
+        String encodedPartition = partitioner.encodePartition(sinkRecord);
+
+        assertEquals("year=2015/month=4/day=2/hour=1/", encodedPartition);
+    }
+
+    @Test
+    public void testNestedRecordFieldTimeExtractor() throws Exception {
+        TimeBasedPartitioner<String> partitioner = new TimeBasedPartitioner<>();
+        Map<String, Object> config = createConfig("nested.timestamp");
+        partitioner.configure(config);
+
+        long timestamp = new DateTime(2015, 4, 2, 1, 0, 0, 0, DateTimeZone.forID(timeZoneString)).getMillis();
+        SinkRecord sinkRecord = createSinkRecordWithNestedTimeField(timestamp);
+
+        String encodedPartition = partitioner.encodePartition(sinkRecord);
+
+        assertEquals("year=2015/month=4/day=2/hour=1/", encodedPartition);
+    }
+
+    @Test
+    public void testRecordTimeExtractor() throws Exception {
+        TimeBasedPartitioner<String> partitioner = new TimeBasedPartitioner<>();
+        Map<String, Object> config = createConfig(null);
+        partitioner.configure(config);
+
+        long timestamp = new DateTime(2015, 4, 2, 1, 0, 0, 0, DateTimeZone.forID(timeZoneString)).getMillis();
+        SinkRecord sinkRecord = createSinkRecord(timestamp);
+
+        String encodedPartition = partitioner.encodePartition(sinkRecord);
+
+        assertEquals("year=2015/month=4/day=2/hour=1/", encodedPartition);
+    }
+
+    private static class BiHourlyPartitioner extends TimeBasedPartitioner<String> {
+        private static long partitionDurationMs = TimeUnit.HOURS.toMillis(2);
+
+        @Override
+        public String getPathFormat() {
+            return "'year'=YYYY/'month'=MMMM/'day'=dd/'hour'=H/";
+        }
+
+        @Override
+        public void configure(Map<String, Object> config) {
+            init(partitionDurationMs, getPathFormat(), Locale.ENGLISH, DATE_TIME_ZONE, config);
+        }
+    }
+
+    private static class HourlySchemaGenerator implements SchemaGenerator<String> {
```

Comment:
Is this really necessary? I noticed the `SchemaGenerator` is only used by the hive module